components/util/logical_geometry.rs

```diff
@@ -85,6 +85,11 @@ impl WritingMode {
             (true, false) => PhysicalSide::Left,
         }
     }
+
+    #[inline]
+    pub fn to_bidi_level(&self) -> u8 {
+        if self.is_bidi_ltr() { 0 } else { 1 }
```

Comment:
Could probably be `(!self.is_bidi_ltr()) as u8`, which may codegen better. (LLVM is not very good at simplifying branches.)